nafflib/optimize.py

```diff
@@ -0,0 +1,172 @@
+import numpy as np
+import numba
+
+
+
+@numba.jit(nopython=True)
```

Comment:
I am not against it, but is numba giving a significant advantage?

Reply:
Yes, factor 2 in speed, suggested by @giadarol